src/modules/joystick/Joystick.h

```diff
@@ -144,6 +144,26 @@ class Joystick : public Object
 		INPUT_TYPE_MAX_ENUM
 	};
 
+	enum PowerState
+	{
+		POWERSTATE_ERROR = -1,   /**< error determining power status */
```

Comment:
I don't think these enums should use `-1` - doing that while putting them in EnumMap/StringMap might not work out well.

Actually an error value shouldn't be part of the enum love uses in the first place I think. I'd rather throw a love exception (which would then be converted to a Lua error) instead, or fall back to `unknown`.